static/js/modules/game/articleRenderer.js

```diff
@@ -5,47 +5,44 @@ export class ArticleRenderer {
     /* frame: DOM element (i.e. through getElementById) to render article in
      * pageCallback: Called upon loading an article, should expect new page and load time
      */
-    constructor(frame, pageCallback, setupPreviews) {
+    constructor(frame, pageCallback, mouseEnter, mouseLeave) {
         this.frame = frame;
         this.pageCallback = pageCallback;
-        this.setupPreviews = setupPreviews;
+        this.mouseEnter = mouseEnter;
+        this.mouseLeave = mouseLeave;
     }
 
     async loadPage(page) {
+        try {
+            const startTime = Date.now();
+            const body = await getArticle(page);
 
-        const body = await getArticle(page);
-
-        this.frame.innerHTML = body["text"]["*"];
-
-        // Create title
-        let titleEl = document.createElement("div");
-        titleEl.innerHTML = "<h1><i>" + body["title"] + "</i></h1>";
-        this.frame.insertBefore(titleEl, this.frame.firstChild);
-
-
-        hideElements(this.frame);
-        // disableFindableLinks(this.frame);
-        stripNamespaceLinks(this.frame);
+            this.frame.innerHTML = body["text"]["*"];
 
+            // Create title
+            let titleEl = document.createElement("div");
+            titleEl.innerHTML = "<h1><i>" + body["title"] + "</i></h1>";
+            this.frame.insertBefore(titleEl, this.frame.firstChild);
 
-        this.frame.querySelectorAll("a, area").forEach((el) => {
-            // Arrow function to prevent this from being overwritten
-            el.onclick = (e) => this.handleWikipediaLink(e);
-        });
 
-        window.scrollTo(0, 0);
+            hideElements(this.frame);
+            // disableFindableLinks(this.frame);
+            stripNamespaceLinks(this.frame);
 
-        return body["title"]
-    }
```

Comment:
I think many of the changes here is just a refactor to remove the extra loadPageWrapper we had before?

Reply:
Yep yep